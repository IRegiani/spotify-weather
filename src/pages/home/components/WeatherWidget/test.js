import React from 'react';
import { shallow } from 'enzyme';

// eslint-disable-next-line no-unused-vars
import Weather, { generateCityComponent } from './index';

jest.mock('react-router-dom');

describe('Weather Widget', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<Weather setWeatherInfo={jest.fn()} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test('renders city text field', () => {
    expect(generateCityComponent({}, {})).toMatchSnapshot();
    // WIP: expect wrapper to contain input
  });

  test('renders city text field', () => {
    expect(generateCityComponent({}, {})).toMatchSnapshot();
    // WIP: expect wrapper to contain input
  });

  test.todo('renders loading status');
});
