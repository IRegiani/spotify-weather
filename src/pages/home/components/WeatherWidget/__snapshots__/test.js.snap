// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Widget renders city text field 1`] = `
<WithStyles(ForwardRef(Tooltip))
  arrow={true}
  leaveDelay={500}
  placement="right"
  title="Click to choose another city"
>
  <WithStyles(ForwardRef(InputBase))
    inputProps={
      Object {
        "aria-label": "naked",
        "style": Object {
          "textAlign": "center",
        },
      }
    }
    onChange={[Function]}
    onClick={[Function]}
    onKeyPress={[Function]}
    type="text"
  />
</WithStyles(ForwardRef(Tooltip))>
`;

exports[`Weather Widget renders city text field 2`] = `
<WithStyles(ForwardRef(Tooltip))
  arrow={true}
  leaveDelay={500}
  placement="right"
  title="Click to choose another city"
>
  <WithStyles(ForwardRef(InputBase))
    inputProps={
      Object {
        "aria-label": "naked",
        "style": Object {
          "textAlign": "center",
        },
      }
    }
    onChange={[Function]}
    onClick={[Function]}
    onKeyPress={[Function]}
    type="text"
  />
</WithStyles(ForwardRef(Tooltip))>
`;

exports[`Weather Widget renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
  variant="outlined"
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Skeleton))
      className="makeStyles-skeletonCircle-4"
      height={45}
      variant="circle"
      width={45}
    />
    <WithStyles(ForwardRef(Skeleton))
      className="makeStyles-skeletonText-5"
      variant="text"
      width={15}
    />
    <WithStyles(ForwardRef(Skeleton))
      className="makeStyles-skeletonText-5"
      variant="text"
      width={55}
    />
    <WithStyles(ForwardRef(Tooltip))
      arrow={true}
      leaveDelay={500}
      placement="right"
      title="Click to choose another city"
    >
      <WithStyles(ForwardRef(InputBase))
        className="makeStyles-input-6"
        inputProps={
          Object {
            "aria-label": "naked",
            "style": Object {
              "textAlign": "center",
            },
          }
        }
        onChange={[Function]}
        onClick={[Function]}
        onKeyPress={[Function]}
        type="text"
      />
    </WithStyles(ForwardRef(Tooltip))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
